qcodes/instrument_drivers/Keysight/N9030B.py

```diff
@@ -0,0 +1,601 @@
+            docstring="start frequency for the sweep"
+        )
+
+        self.add_parameter(
+            name="stop",
+            get_cmd=":SENSe:FREQuency:STOP?",
+            set_cmd=self._set_stop,
+            get_parser=float,
+            vals=Numbers(self._min_freq + 10, self._max_freq),
+            docstring="stop frequency for the sweep"
+        )
+
+        self.add_parameter(
+            name="center",
+            get_cmd=":SENSe:FREQuency:CENTer?",
+            set_cmd=self._set_center,
+            get_parser=float,
+            vals=Numbers(self._min_freq + 5, self._max_freq - 5),
+            docstring="Sets and gets center frequency"
+        )
+
+        self.add_parameter(
+            name="span",
+            get_cmd=":SENSe:FREQuency:SPAN?",
+            set_cmd=self._set_span,
+            get_parser=float,
+            vals=Numbers(10, self._max_freq - self._min_freq),
+            docstring="Changes span of frequency"
+        )
+
+        self.add_parameter(
+            name="npts",
+            get_cmd=":SENSe:SWEep:POINts?",
+            set_cmd=self._set_npts,
+            get_parser=int,
+            vals=Ints(1, 20001),
+            docstring="Number of points for the sweep"
+        )
+
+        self.add_parameter(
+            name="sweep_time",
+            label="Sweep time",
+            get_cmd=":SENSe:SWEep:TIME?",
+            set_cmd=":SENSe:SWEep:TIME {}",
+            get_parser=float,
+            unit="s",
+            docstring="gets sweep time"
+        )
+
+        self.add_parameter(
+            name="auto_sweep_time_enabled",
+            initial_value=False,
+            get_cmd=":SENSe:SWEep:TIME:AUTO?",
+            set_cmd=self._enable_auto_sweep_time,
+            val_mapping=create_on_off_val_mapping(on_val="ON", off_val="OFF"),
+            docstring="enables auto sweep time"
+        )
+
+        self.add_parameter(
+            name="auto_sweep_type_enabled",
+            initial_value=False,
+            get_cmd=":SENSe:SWEep:TYPE:AUTO?",
+            set_cmd=self._enable_auto_sweep_type,
+            val_mapping=create_on_off_val_mapping(on_val="ON", off_val="OFF"),
+            docstring="enables auto sweep type"
+        )
+
+        self.add_parameter(
+            name="sweep_type",
+            get_cmd=":SENSe:SWEep:TYPE?",
+            set_cmd=self._set_sweep_type,
+            val_mapping={
+                "fft": "FFT",
+                "sweep": "SWE",
+            },
+            docstring="Sets up sweep type. Possible options are 'fft' and "
+                      "'sweep'."
+        )
+
+        self.add_parameter(
+            name="freq_axis",
+            label="Frequency",
+            unit="Hz",
+            start=self.start,
+            stop=self.stop,
+            npts=self.npts,
+            vals=Arrays(shape=(self.npts.get_latest,)),
+            parameter_class=FrequencyAxis,
+            docstring="Sets frequency axis for the sweep."
```

Comment:
This parameter should not be settable since its set via the start stop and npoints

Reply:
Updated the docstring. Parameter doesn't actually sets but creates the axis from start stop and npts